### PR TITLE
Added Longest Common Subsequence in Haskell

### LINKED ISSUE #576: Add Longest Common Subsequence in Haskell

### PR DESCRIPTION
**Pull request by auroq**
Congrats on taking the first step to contributing to the [Sample Programs in Every Language][1] repository!

For simplicity, please make sure that your pull request includes one and only one sample program and language.

## Please Complete the Following

- [ ] I would **NOT** like credit for my contribution in the article (credit will be included otherwise)
- [ ] I plan to write the article for this code snippet
- [x] I named the pull request using `<Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [ ] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])
- [x] I fixed #576 

## Notes

Feel free to put whatever you want here.

[1]: https://therenegadecoder.com/code/sample-programs-in-every-language/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/CONTRIBUTING.md#plagiarism

### CODE REVIEW

File: archive/h/haskell/lcs.hs [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  | otherwise             = l2
+
+
+main :: IO ()
+main = do
+  args <- getArgs
+  let l1 = read $ head args :: [Int]
+  let l2 = read $ head $ tail args :: [Int]
+  if length args /= 2 then
+    error "You need to pass two lists of which to produce the lcs. For example ./lcs \"[1, 4, 5, 3, 15, 6]\" \"[1, 7, 4, 5, 11, 6]\"\n"
```
Comment (jrg94) [inline archive/h/haskell/lcs.hs]: Can we change this so the program just accepts comma-separated lists like "1, 3, 5, 7" and "2, 3, 7"?

**Comment by auroq** [comment]
@jrg94 I found the bug you mentioned. I had hardcoded lists in for testing and forgot to switch it back to user input. It is fixed now.
I also changed it to accept lists in the format "1, 2, 3"

File: archive/h/haskell/lcs.hs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+main :: IO ()
+main = do
+  args <- getArgs
+  let l1 = stringToList $ head args :: [Int]
+  let l2 = stringToList $ head $ tail args :: [Int]
+  if length args /= 2 then
+    error "You need to pass two lists of which to produce the lcs. For example ./lcs \"1, 4, 5, 3, 15, 6\" \"1, 7, 4, 5, 11, 6\"\n"
+  else
+    let message    = "\nThe longest common subsequence of " ++ (show l1) ++ " and " ++ (show l2) ++ " is "
```
Comment (jrg94) [inline archive/h/haskell/lcs.hs]: Excellent! For consistency, can you have this output the resulting array only?

**Comment by auroq** [comment]
@jrg94 I changed it to output just the result. I initially did it the other way to follow consistency with fibonacci.hs. We might want to change that one as well so they are all the same.

Haskell is a beautiful language when written properly, even if you're not familiar with it. It's one thing that I enjoyed when starting to learn it.

**Comment by jrg94** [comment]
I'm in the process of standardizing our documentation. Fibonacci has a special print pattern because we let anyone who contributes write their own output routine. Roman numeral and LCS should just output the result according to the rules. So, your fibonacci should be fine as long as it outputs fib like:

```console
1: 1
2: 1
3: 2
4: 3
5: 5
```

**Pull request merged**